libs/components/forms/src/lib/modules/required-state/required-state.directive.ts

```diff
@@ -1,27 +1,80 @@
-import { Directive, Input, booleanAttribute, inject } from '@angular/core';
+import {
+  AfterViewInit,
+  Directive,
+  EventEmitter,
+  Injector,
+  Input,
+  OnDestroy,
+  Output,
+  booleanAttribute,
+  inject,
+} from '@angular/core';
 import { NgControl, Validators } from '@angular/forms';
 
+import { Subject, takeUntil } from 'rxjs';
+
 /**
  * A host directive used to capture the "required" state of the host's form control.
  * @internal
  */
 @Directive({
   standalone: true,
 })
-export class SkyRequiredStateDirective {
+export class SkyRequiredStateDirective implements OnDestroy, AfterViewInit {
   @Input({ transform: booleanAttribute })
-  public required = false;
+  public set required(value: boolean) {
+    this.#_required = value;
+    this.isRequired();
+  }
+
+  public get required(): boolean {
+    return this.#_required;
+  }
+
+  @Output()
+  public requiredStateChange = new EventEmitter();
 
   /**
    * Whether the `required` attribute is set to `true`, or the control includes
    * the `Validators.required` validator.
    */
   public isRequired(): boolean {
-    return (
+    const requiredState =
       this.required ||
-      !!this.#ngControl?.control?.hasValidator(Validators.required)
-    );
+      !!this.#ngControl?.control?.hasValidator(Validators.required);
+
+    if (requiredState !== this.#currentRequiredState) {
+      this.#currentRequiredState = requiredState;
+      this.requiredStateChange.emit(requiredState);
+    }
+
+    return requiredState;
   }
 
-  readonly #ngControl = inject(NgControl, { optional: true, self: true });
+  #_required = false;
+
+  #currentRequiredState = false;
+  #injector = inject(Injector);
+  #ngControl?: NgControl | null;
+  readonly #ngUnsubscribe = new Subject<void>();
+
+  public ngOnDestroy(): void {
+    this.#ngUnsubscribe.next();
+    this.#ngUnsubscribe.complete();
+  }
+
+  public ngAfterViewInit(): void {
+    this.#ngControl = this.#injector.get(NgControl, undefined, {
```

Comment:
Doing this in the `AfterViewInit` was needed to get around a circular reference when the ControlValueAccessor gave the `NG_VALIDATORS` provider in its providers array. We do this in a couple other components. Some do it on `ngOnInit` but `statusChanges` is not ready until `ngAfterViewInit`.